org.adempiere.base.process/src/org/compiere/process/InOutGenerate.java

```diff
@@ -290,7 +291,17 @@ private String generate (PreparedStatement pstmt)
 							logInfo.append(" (set to 0)");
 						}
 						//	Adjust On Hand
-						onHand = onHand.subtract(unconfirmedShippedQty);
+						StringBuilder where3 = new StringBuilder ("EXISTS (SELECT * FROM M_InOut io WHERE io.M_InOut_ID=M_InOutLine.M_InOut_ID AND io.IsSOTrx = 'Y' AND io.DocStatus IN ('DR','IN','IP','WC')")
```

Comment:
Shouldn't include DR and IN

Reply:
Hi @hengsin,
when we created this patch two years ago we took a look at the other statements dealing with unconfirmed inouts and took the same list of doc states. Perhaps both states are wrong there, too. Can you have a look at it?

Reply:
hi @mschnbeck , can you point me where to check for that ?

Reply:
hi @mschnbeck , my mistake, the status clause is fine. However, with your changes, the existing unconfirmedShippedQty (and the query to get it) is make redundant and should be removed from the code.

Reply:
Hi @hengsin, I don't think that the unconfirmedShippedQty is redundant. This is the quantity we already have on shipments not yet completed for _this_ order and therefore don't have to create a shipment again. Thus to calculate how much remains to be shipped. The totalunconfirmedShippedQty however is the quantity which seems to be stocked but really isn't available anymore, because it's on not yet completed shipments for this or other orders.

What could be optmized is, not to look for the totalunconfirmedShippedQty if the toDeliver is already zero or set to zero.